<a href="https://colab.research.google.com/github/Martishamao/Formula-1-Analysis/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project 2
- Martisha Owens
- 9/29/2022

#DataSet 1


##1. Souce of data
- Formula 1 World Championship (1950 - 2022)
- https://www.kaggle.com/datasets/rohanrao/formula-1-world-championship-1950-2020?select=results.csv

##2. Brief description of data
- The FIA Formula One World Championship results since 1950 to 2022. The dataset also consists of all information on the Formula 1 races, drivers, constructors, qualifying, circuits, lap times, and pit stops.

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [21]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

filename = '/content/drive/MyDrive/Coding Dojo/Project 2/results.csv'
df1 = pd.read_csv(filename)
df1.tail()

resultId  raceId  driverId  constructorId number  grid position  \
25655     25661    1086       825            210     20    13       16   
25656     25662    1086       848              3     23    17       17   
25657     25663    1086       849              3      6    19       18   
25658     25664    1086       852            213     22    16       19   
25659     25665    1086       822             51     77     8       20   

      positionText  positionOrder  points  laps time milliseconds fastestLap  \
25655           16             16     0.0    69   \N           \N         37   
25656           17             17     0.0    69   \N           \N         43   
25657           18             18     0.0    69   \N           \N         60   
25658           19             19     0.0    68   \N           \N         58   
25659           20             20     0.0    65   \N           \N         60   

      rank fastestLapTime fastestLapSpeed  statusId  
25655   15       1:23.511         188.856        11  
25656   12       1:23.047         189.911        11  
25657    8       1:22.478         191.221        11  
25658   16       1:23.538         188.795        12  
25659   19       1:24.002         187.752       131

> Formula 1 Data Columns
- resultId
- raceId, races.csv
- driverId, drivers.csv
- constructorId, Car Constructor, constructors.csv
- number, Car Number
- grid, Position in starting grid
- position, Final position
- positionText, Final position in text
- positionOrder, Final rank
- points
- laps
- time *(Remove)*
- milliseconds *(Remove)*
- fastestLap *(Remove)*
- rank
- fastestLapTime *(Remove)*
- fastestLapSpeed
- statusId, position status, status.csv

##3. What is the target?
- The target column will be the position column. Target question would be predicting what driver will make podium. 

##4. Is this a classification or regression problem?
- The positions column is Ordinal thus this is a classification problem. 

##5&6. How many features and rows?
- Removing any time series columns.
- resultId was also removed because it is redundant.

In [29]:
df1c = df1.drop(columns = ['resultId','time','milliseconds','fastestLap','fastestLapTime'])
df1c.tail(10)

raceId  driverId  constructorId number  grid position positionText  \
25650    1086       840            117     18    14       11           11   
25651    1086       842            213     10     0       12           12   
25652    1086       855             51     24    12       13           13   
25653    1086       854            210     47    15       14           14   
25654    1086       817              1      3     9       15           15   
25655    1086       825            210     20    13       16           16   
25656    1086       848              3     23    17       17           17   
25657    1086       849              3      6    19       18           18   
25658    1086       852            213     22    16       19           19   
25659    1086       822             51     77     8       20           20   

       positionOrder  points  laps rank fastestLapSpeed  statusId  
25650             11     0.0    69    7         191.317        11  
25651             12     0.0    69   14         189.564        11  
25652             13     0.0    69    5         192.268        11  
25653             14     0.0    69   13         189.674        11  
25654             15     0.0    69   17         188.533        11  
25655             16     0.0    69   15         188.856        11  
25656             17     0.0    69   12         189.911        11  
25657             18     0.0    69    8         191.221        11  
25658             19     0.0    68   16         188.795        12  
25659             20     0.0    65   19         187.752       131

In [30]:
df1c.isna().sum()

raceId             0
driverId           0
constructorId      0
number             0
grid               0
position           0
positionText       0
positionOrder      0
points             0
laps               0
rank               0
fastestLapSpeed    0
statusId           0
dtype: int64

In [31]:
df1c.duplicated().sum()

0

>No missing or duplicated data.

In [20]:
df1c.shape

(25660, 13)

> There are 13 columns and 25,600 rows of data.

##7. What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
> From a first glance, this dataset contains the id number and requires a different dataset to reference. This can also be resolved with a OneHotEncoder.

#DataSet 2

##1. Souce of data
- Netflix popular movies dataset
- https://www.kaggle.com/datasets/narayan63/netflix-popular-movies-dataset 

##2. Brief description of data
- This data is a list of tv shows and movies that are available on Netflix. 
- The data also includes the movie/tv title, year, certificate (TV rating), duration, genre, rating on IMDB, description of movies, cast, and number of voters.

In [27]:
filename = '/content/drive/MyDrive/Coding Dojo/Project 2/n_movies.csv'
df2 = pd.read_csv(filename)
df2.head()

title         year certificate duration  \
0               Cobra Kai     (2018– )       TV-14   30 min   
1               The Crown     (2016– )       TV-MA   58 min   
2        Better Call Saul  (2015–2022)       TV-MA   46 min   
3           Devil in Ohio       (2022)       TV-MA  356 min   
4  Cyberpunk: Edgerunners     (2022– )       TV-MA   24 min   

                          genre  rating  \
0         Action, Comedy, Drama     8.5   
1     Biography, Drama, History     8.7   
2                  Crime, Drama     8.9   
3        Drama, Horror, Mystery     5.9   
4  Animation, Action, Adventure     8.6   

                                         description  \
0  Decades after their 1984 All Valley Karate Tou...   
1  Follows the political rivalries and romance of...   
2  The trials and tribulations of criminal lawyer...   
3  When a psychiatrist shelters a mysterious cult...   
4  A Street Kid trying to survive in a technology...   

                                               stars    votes  
0  ['Ralph Macchio, ', 'William Zabka, ', 'Courtn...  177,031  
1  ['Claire Foy, ', 'Olivia Colman, ', 'Imelda St...  199,885  
2  ['Bob Odenkirk, ', 'Rhea Seehorn, ', 'Jonathan...  501,384  
3  ['Emily Deschanel, ', 'Sam Jaeger, ', 'Gerardo...    9,773  
4  ['Zach Aguilar, ', 'Kenichiro Ohashi, ', 'Emi ...   15,413

##3. What is the target?
- The target is the tv/movie IMDB rating.

##4. Is this a classification or regression problem?
- This is a regression problem, which will predict the rating on a continuous scale from 0 to 10.

##5&6. How many features and rows?


In [33]:
df2c = df2.copy()

In [38]:
df2c.isna().sum()

title             0
year            527
certificate    3453
duration       2036
genre            73
rating         1173
description       0
stars             0
votes          1173
dtype: int64

In [46]:
print(f'Missing ratings data: {round(1173/9957*100,2)}%')

Missing ratings data: 11.78%


In [34]:
df2c.duplicated().sum()

0

In [39]:
df2c.shape

(9957, 9)

> There are no duplicated data but there are quite a few missing data including the rating columns. If these missing rating data were removed from the dataset, we will still have more than 1,000 rows. The data has 9 columns and 9957 rows.

##7. What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
> There are multiple actors listed under cast. We might have to split this into multiple columns. About 12% of the ratings data is missing which is a signification amount. Not sure how to incorporate the description into the predictions just yet.

#Other
- Apparel Exports to US, https://data.world/makeovermonday/2020w43-apparel-exports-to-us
- New York City Airbnb Open Data, https://www.kaggle.com/datasets/dgomonov/new-york-city-airbnb-open-data